core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -1310,6 +1413,35 @@ public Command workerHeartbeat(long workerId, Map<String, Long> capacityBytesOnT
     // by the LostWorkerDetectionHeartbeatExecutor
     worker.updateLastUpdatedTimeMs();
 
+    if (mWorkerRegisterToAllMasters && mPrimarySelector.getState() == NodeState.STANDBY) {
```

Comment:
on standby master, heartbeat can come and be processed from a journal.
e.g. 
A journal that creates a block is processed after a worker master heartbeat to remove such block.
This will result in some state inconsistency so we wait here for a best effort first before we ignore the block change in heartbeat